docs/operators/fsInputOperator.md

```diff
@@ -0,0 +1,101 @@
+
+![AbstractFileInputOperator.png](images/fsInput/operatorsClassDiagram.png)
+
+## Properties, Attributes and Ports
+### <a name="props"></a>Properties of AbstractFileInputOperator
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *directory* | absolute path of directory to be scanned | String | Yes | N/A |
+| *scanIntervalMillis* | Interval in milliseconds after which directory should be scanned for new files | int | No | 5000 |
+| *emitBatchSize* | Number of tuples to emit in a batch | int | No | 1000 |
+| *partitionCount* | Desired number of partitions count | int | No | 1 |
+| *maxRetryCount* | Maximum number of times the operator will attempt to process a file | No | 5 |
+| *scanner* | Scanner to scan new files in directory | [DirectoryScanner](#DirectoryScanner) | No | DirectoryScanner |
+
+#### <a name="JDBCStore"></a>Properties of DirectoryScanner
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *filePatternRegexp* | regex to select files from input directory | String | No | N/A |
+
+## Abstract Methods
+void emit(T tuple): Abstract method that emits tuple read from file.
+T readEntity(): Abstract method to read file entity (can be a line or block).
+
+## Partitioning of File Input Operator
+Configure parameter "partitionCount" to define desired number of partitions.
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.prop.partitionCount</name>
+    <value>4</value>
+  </property>
+```
+
+where {OperatorName} is the name of the input operator.
+Above lines will partition operator statically 4 times. Above value can be changed accordingly to change the number of static partitions.
+
+#### Dynamic Partitioning of JDBC Poller
+Implement ```getNewPartitionCount()``` method to do dymanic partitioning.
+
+## Derived Classes
+### 1. AbstractFTPInputOperator
+The class is used to read files from FTP file system.
+#### Properties, Attributes and Ports
+#### <a name="props"></a>Properties of AbstractFTPInputOperator
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *host*| Hostname of ftp server.| String | Yes | N/A |
+| *port*| Port of ftp server.| int | No | 21 (default ftp port) |
+| *userName*| Username which is used for login to the server. | String | No | anonymous |
+| *password*| Password which is used for login to the server. | String | No | gues |
+
+### 2. AbstractParquetFileReader
+Reads Parquet files from input directory using GroupReadSupport. Derived classes need to implement [convertGroup(Group)](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/parquet/AbstractParquetFileReader.html#convertGroup(Group)) method to convert Group to other type.
```

Comment:
Since we are being explicit about what derived classes need to do here, please verify that we do the same for all the other operators described here.